language-server/src/main/scala/scalafix/languageserver/ScalafixService.scala

```diff
@@ -0,0 +1,135 @@
+// In scalafix package to access private[scalafix] methods.
+package scalafix.languageserver
+
+import java.io.PrintStream
+import java.nio.file.Files
+import scala.meta.internal.tokenizers.PlatformTokenizerCache
+import scala.tools.nsc.interpreter.OutputStream
+import scala.{meta => m}
+import scalafix._
+import scalafix.internal.config.LazySemanticdbIndex
+import scalafix.internal.config.ScalafixConfig
+import scalafix.internal.config.ScalafixReporter
+import scalafix.internal.util.EagerInMemorySemanticdbIndex
+import scalafix.lint.LintSeverity
+import scalafix.patch.Patch
+import scalafix.reflect.ScalafixReflect
+import scalafix.rule.RuleCtx
+import scalafix.rule.RuleName
+import scalafix.util.SemanticdbIndex
+import langserver.core.Connection
+import langserver.messages.MessageType
+import langserver.{types => l}
+import metaconfig.ConfDecoder
+import org.langmeta.internal.semanticdb.{schema => s}
+import org.langmeta.io.AbsolutePath
+import org.langmeta.io.RelativePath
+
+case class ScalafixResult(path: RelativePath, diagnostics: Seq[l.Diagnostic])
+
+class ScalafixService(cwd: AbsolutePath,
+                      out: OutputStream,
+                      connection: Connection) {
+  def configFile: Option[m.Input] = ScalafixConfig.auto(cwd)
```

Comment:
does this mean the file is reloaded from disk at every execution of scalafix? If this is not an expensive operation it would save the burden of syncing the file (so what I said above may not be true)